File: MealSaver/src/main/java/com/marcu/mealsaver/Dto/FoodDTO.java
package com.marcu.mealsaver.Dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class FoodDTO {

    @NotNull(message = "Id is required")
    private Long id;

    @NotBlank(message = "Name is required")
    private String name;

    @NotNull(message = "Size is required")
    @Min(value = 1, message = "Size must be at least 1")
    private Integer size;

    @NotNull(message = "Expiration date is required")
    private Date expirationDate;

    @NotBlank(message = "Username is required")
    private String username;
}

File: meal_saver_phone/meal_saver_phone/lib/models/food_dto.dart
class FoodDTO {
  final int id;
  final String name;
  final int size;
  final String expirationDate;

  FoodDTO({
    required this.id,
    required this.name,
    required this.size,
    required this.expirationDate,
  });

  factory FoodDTO.fromJson(Map<String, dynamic> json) {
    return FoodDTO(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      size: json['size'] ?? 0,
      expirationDate: json['expirationDate'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'size': size,
      'expirationDate': expirationDate,
    };
  }
}
